src/types/instances.ts

```diff
@@ -0,0 +1,123 @@
+import { ethers } from 'ethers';
+
+import { CreateProposalParams, CreateZDAOParams } from './params';
+import { Choice, ProposalId, VoteId, zNA } from './primitives';
+import {
+  ProposalProperties,
+  TokenMetaData,
+  Transaction,
+  Vote,
+  zDAOAssets,
+  zDAOProperties,
+} from './structures';
+
+export interface SDKInstance {
+  /**
+   * Get all the list of zDAO
+   * @returns list of zNA
+   */
+  listZDAOs(): Promise<zNA[]>;
+
+  /**
+   * Create an zDAO instance by zNA
+   * @param zNA zNA address
+   * @returns created zDAO instance
+   * @exception throw Error if zNA does not exist
+   */
+  getZDAOByZNA(zNA: zNA): Promise<zDAO>;
+
+  /**
+   * Check if zDAO exists
+   * @param zNA zNA address
+   * @returns true if zNA exists
+   */
+  doesZDAOExist(zNA: zNA): Promise<boolean>;
+
+  /**
+   * Create zDAO from parameters for test
+   * @param packaged parameters of zDAO
+   * @exception throw Error if zNA already exists
+   * @exception throw Error if owners is empty
+   * @exception throw Error if title is empty
+   */
+  createZDAOFromParams(param: CreateZDAOParams): Promise<void>;
```

Comment:
I see now how you implemented this. Very interesting.
Could you please create an additional method `getZDAOFromParams` which takes the following parameters:

- ens id (real)
- gnosis safe

The 'dao id' can be faked.

ENS is actually contacted for the ENS related data.

This function returns a `Promise<zDAO>`